### PR TITLE
homework week 2

### PR DESCRIPTION
**Pull request by melo4yxa77796**
The search engine will have 3 routes:

GET /search
GET /documents/:id
POST /search

### CODE REVIEW

File: nodejs:week2/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+  const documents = getDocuments();
+
+  if (!query) {
+    return res.json([]); // Return an empty array if no query provided
+  }
+
+  const filteredDocuments = documents.filter(doc =>
+    doc.value && doc.value.includes(query)
```
Comment (urbanogilson) [inline nodejs:week2/week2/app.js]: Your filter is [case-sensitive](https://www.lifewire.com/what-does-case-sensitive-mean-2625824). Would be nice to make case-insensitive

File: nodejs:week2/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  if (fields) {
+    filteredDocuments = filteredDocuments.filter(doc => {
+      return Object.entries(fields).every(([key, value]) => doc[key]&&doc[key] === value);
+    });
+  }
+
+  // If q is provided, filter by the query
+  if (query) {
+    filteredDocuments = filteredDocuments.filter(doc =>
+      doc.value && doc.value.includes(query)
```
Comment (urbanogilson) [inline nodejs:week2/week2/app.js]: This filter is duplicated. An improvement would be to create a function to filter the documents, avoiding code duplication in `GET` and `POST` for the `search API`

**Review by urbanogilson — COMMENTED** [review]
Good job!

File: nodejs:week2/week2/app.js (hunk 1 above)
Comment (melo4yxa77796) [reply nodejs:week2/week2/app.js]: 👍

File: nodejs:week2/week2/app.js (hunk 2 above)
Comment (melo4yxa77796) [reply nodejs:week2/week2/app.js]: i did it

**Pull request merged**